### PR TITLE
[gmp] Reset m_draining when drain complete and on reset. JB#51577

### PR DESCRIPTION
**Pull request by abranson**
Looping videos reuse the same codec, which was stuck in the draining state at the end of the last loop.

### CODE REVIEW

File: gmp-droid.cpp [hunk 1]
```diff
@@ -266,6 +266,8 @@ class DroidVideoDecoder : public GMPVideoDecoder
     if (m_codec) {
       ResetCodec ();
     }
+    m_draining = false;
+    m_resetting = false;
```
Comment (llewelld) [inline gmp-droid.cpp:270]: Can the `m_resetting = true;` on line 263 be removed, or is there some multithreading magic going on here?

**Review by llewelld — APPROVED** [review]
Just one small comment, but this looks sane to me, and also does the trick. Thanks!

File: gmp-droid.cpp (hunk 1 above)
Comment (abranson) [reply gmp-droid.cpp:270]: That's the only place it's set to true I think. I need that boolean as ResetCodec can take a while to finish and I have to make sure that doesn't re-enter.

Comment (llewelld) [reply gmp-droid.cpp:270]: Okay, that makes sense.

**Review by mlehtima — APPROVED** [review]

File: gmp-droid.cpp (hunk 1 above)
Comment (llewelld) [reply gmp-droid.cpp:270]: Perhaps you should use something like `std::atomic<bool>` for `m_draining` to avoid optimisation+threading issues?

**Comment by monich** [comment]
Unsynchronized operations on variables looks very fragile if there are threads involved (and I'm sure threads are involved).

**Pull request merged**